src/cli/commands/init.js

```diff
@@ -22,33 +21,26 @@ module.exports = {
       .option('emptyRepo', {
         alias: 'e',
         type: 'boolean',
-        describe: "Don't add and pin help files to the local storage"
+        describe: 'Don\'t add and pin help files to the local storage'
       })
   },
 
   handler (argv) {
-    const path = utils.getRepoPath()
+    const path = getRepoPath()
 
     print(`initializing ipfs node at ${path}`)
 
-    const node = new IPFS({
+    return IPFS.createNodePromise({
       repo: new Repo(path),
       init: false,
       start: false
-    })
```

Comment:
Does `createNodePromise` resolve when `start: false`?

Reply:
yes, ready is always emitted if an error doesn't happen 
https://github.com/ipfs/js-ipfs/blob/feat/better-errors-cli/src/core/boot.js#L53